docs/source/tutorials/arch_modeling/user_defined_temp_tutorial.rst

```diff
@@ -0,0 +1,167 @@
+Integrating Custom Verilog Modules with user_defined_templates.v
+================================================================
+Introduction and Setup
+~~~~~~~~~~~~~~~~~~~~~~
+**In this tutorial, we will**
+   - Provide motivation for generating the user_defined_templates.v verilog file
+   - Go through a generated user_defined_templates.v file to demonstrate how to use it
+Through this example, we will motivate and show how to use the ``user_defined_templates.v`` file.
+
+For this examaple, we are using a modified version of the hard adder task that comes with OpenFPGA.
+To follow along, go to the root directory of OpenFPGA and enter: 
+
+.. code-block:: XML
+
+   vi openfpga_flow/openfpga_arch/k6_frac_N10_adder_chain_40nm_openfpga.xml 
+
+Go to line 187 and replace it with:
+
+.. code-block:: XML
+
+   <circuit_model type="hard_logic" name="ADDF" prefix="ADDF" is_default="true" spice_netlist="${OPENFPGA_PATH}/openfpga_flow/openfpga_cell_library/spice/adder.sp"     verilog_netlist="">
+
+Motivation
+~~~~~~~~~~
+From the OpenFPGA root directory run the command
+
+.. code-block:: XML
+
+   python3 openfpga_flow/scripts_run_fpga_task.py fpga_verilog/adder/hard_adder --debug --show_thread_logs
+
+Running this command should fail and produce these errors:
+
+.. code-block:: XML
```

Comment:
You may use bash as the language of code-block.
https://build-me-the-docs-please.readthedocs.io/en/latest/Using_Sphinx/ShowingCodeExamplesInSphinx.html#pygments-lexers